### DISCUSSION TITLE
Need help with config - [Errno 98] error while attempting to bind on address

### DISCUSSION THREAD

**Discussion by k9barry** [YoRyan/mailrise#29] [General]
I am struggling to get mailrise to work with traefik in docker.  I am trying to use my domain (changed to mydomain.com for security) and I have pointed my DNS MX to mydomain.com (not sure if the MX record is needed).  I try to forward an email from outside my server to apprise@mydomain.com but the email bounces. 

The error from the log shows [Errno 98] error while attempting to bind on address port 8025.  I have checked docker ps and none of my other services are using that port.  

Any help will be appreciated!

-----------------------------------------------------------

My config file is:

configs:
    apprise@mydomain.com:
        urls:
            - ntfy://ntfy.mydomain.com/TEST
            - pover://user-token-redacted/token-redacted

My docker-compose file is:
mailrise:
    image: yoryan/mailrise
    container_name: mailrise
    labels:
      - com.centurylinklabs.watchtower.enable=true
      - traefik.tcp.routers.mailrise.rule=HostSNI(`*`)
      - traefik.tcp.routers.mailrise.tls=true
      - traefik.tcp.routers.mailrise.tls.certresolver=lets-encrypt
      - traefik.tcp.routers.mailrise.tls.domains[0].main=mailrise.mydomain.com
      - traefik.tcp.routers.mailrise.tls.domains[0].sans=mydomain.com
      - traefik.tcp.routers.mailrise.entrypoints=mailsecure
    volumes:
      - $PWD/data_docker/mailrise/mailrise.conf:/etc/mailrise.conf:ro
    restart: unless-stopped
    networks:
      - internal
      - web
      - 
My Traefik static file is:

[entryPoints]
  [entryPoints.web]
    address = ":80"

  [entryPoints.web.http.redirections.entryPoint]
    to = "websecure"
    scheme = "https"

  [entryPoints.websecure]
    address = ":443"

  [entryPoints.mailsecure]
    address = ":465"`

Logs from my mailrise docker listed below:
`Verbose l[2022-11-28 20:35:33] INFO:mailrise.skeleton:Loaded configuration with 1 recipient(s)
[2022-11-28 20:35:33] DEBUG:asyncio:Using selector: EpollSelector
[2022-11-28 20:35:38] CRITICAL:mailrise.skeleton:Failed to start aiosmtpd controller: [Errno 98] error while attempting to bind on address ('::', 8025, 0, 0): address already in use
Traceback (most recent call last):
  File "/home/mailrise/.local/bin/mailrise", line 8, in <module>
    sys.exit(run())
             ^^^^^
  File "/home/mailrise/.local/lib/python3.11/site-packages/mailrise/skeleton.py", line 164, in run
    main(sys.argv[1:])
  File "/home/mailrise/.local/lib/python3.11/site-packages/mailrise/skeleton.py", line 145, in main
    controller.start()
  File "/home/mailrise/.local/lib/python3.11/site-packages/aiosmtpd/controller.py", line 210, in start
    raise self._thread_exception
  File "/home/mailrise/.local/lib/python3.11/site-packages/aiosmtpd/controller.py", line 176, in _run
    srv: AsyncServer = self.loop.run_until_complete(self.server_coro)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 650, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1511, in create_server
    raise OSError(err.errno, 'error while attempting '
OSError: [Errno 98] error while attempting to bind on address ('::', 8025, 0, 0): address already in use`

**Comment by YoRyan** [comment]
I don't have any specific advice to offer, unfortunately. Maybe run a `docker ps -a` and look for any stopped containers you can clean up. Are you able to run Mailrise without compose?

```bash
docker run -it --rm -v $PWD/data_docker/mailrise/mailrise.conf:/etc/mailrise.conf:ro yoryan/mailrise
```

**Reply by k9barry** [reply]
Thank you for the response.   I will try to run mailrise without compromise and see if that advances me further.

**Reply by k9barry** [reply]
Getting the same errors when using docker run.  Thank you for the suggestions.

**Reply by YoRyan** [reply]
I would try once more with an absolute minimal config:
```yaml
configs:
  test:
    urls:
      - json://localhost
```
```
$ docker run -it --rm -v /path/to/my/mailrise.conf:/etc/mailrise.conf:ro yoryan/mailrise -v /etc/mailrise.conf
[2022-12-05 20:03:09] INFO:mailrise.skeleton:Loaded configuration with 1 recipient(s)
[2022-12-05 20:03:09] INFO:mail.log:Available AUTH mechanisms: LOGIN(builtin) PLAIN(builtin)
- snip -
```
If this doesn't work, I'm afraid I'm out of ideas. Your Docker daemon could be in some kind of weird state.

It's a shot in the dark, but you could also try the `yoryan/mailrise:latest-buster` image, which is based on Debian Buster instead of Bullseye.
